options/options_helper.cc

```diff
@@ -253,6 +254,25 @@ std::unordered_map<std::string, CompressionType>
         {"kZSTD", kZSTD},
         {"kZSTDNotFinalCompression", kZSTDNotFinalCompression},
         {"kDisableCompressionOption", kDisableCompressionOption}};
+
+std::unordered_map<std::string, CompactionReason>
```

Comment:
I think we shouldn't follow `GetStringFromCompressionType()`'s implementation pattern - just following the declaration pattern (declaring `GetStringFromCompactionReason()` in a file under an "include/" directory) is enough. `CompactionReason` is not an option so its logic doesn't really belong here. We also don't do lookup based on string so the map isn't necessary.

Also if you want to make analogous changes for the other enums in "include/rocksdb/listener.h", that'd be helpful (e.g., `TableFileCreationReason`).

Reply:
I'm not sure where it should be defined. I thought of putting the definition in `compaction.h`, but then the other enums' definitions wouldn't fit in there. What file would you suggest for that?

Reply:
Looks like everything declared in `include/rocksdb/listener.h` so far is defined inline. We could either continue that pattern or create a `db/listener.cc` file for these definitions. Or, if you have the declarations in `include/rocksdb/convenience.h`, then the definitions should go in `db/convenience.cc`.